src/basic/cgroup-util.c

```diff
@@ -1977,6 +1977,14 @@ int cg_slice_to_path(const char *unit, char **ret) {
                 _cleanup_free_ char *escaped = NULL;
                 char n[dash - p + sizeof(".slice")];
 
+#ifdef FUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION
+                /* msan doesn't instrument stpncpy, so it thinks
+                 * n is later used unitialized:
+                 * https://github.com/google/sanitizers/issues/926
+                 */
+                zero(n);
```

Comment:
As I was saying I'm not sure the code should be patched up to make `msan` happy. If it was decided to do that anyway, it would probably be better to fix https://github.com/systemd/systemd/issues/8474 first before adding anything else. Currently `fuzz-unit-file` fails under `msan` whatever input it gets.

Reply:
It's one thing, I think, to really update the code, and another to add a clearly-delimited `#ifdef`. It's pretty clear to anyone looking at the code that it's a temporary work-around, and we'll just remove it when msan is fixed.

I'm looking into #8474, it fails here too.

Reply:
I've been patching up `systemd` to make it work under `valgrind` for some time. What was supposed to be just an `ifdef` has become a complete mess, which is probably not too ugly because `valgrind` allows suppressing known false positives without touching the code.

I was looking for something similar to https://clang.llvm.org/docs/AddressSanitizer.html#suppressing-reports-in-external-libraries which would work with `msan`, but I haven't found anything yet. Do you know if it is possible to suppress reports without changing the code?

Reply:
As was mentioned in https://github.com/google/oss-fuzz/issues/1247#issuecomment-373525408, `asan` and `ubsan` builds aren't blocked at all, so I'm not sure why these false positives should be fixed in the first place. Everybody knows they are false positives and they will be closed as soon as everything has started to work properly.

Reply:
fuzzers stop on the first "error". Any code paths that would be reachable through this point are not going to be tested. This is why I prefer to silence this error, so that we can get additional feedback. I think that a minor `#ifdef` block like that is not problematic at all.

Reply:
I think these blocks are problematic in the sense that they currently affect not only `msan` builds but also `ubsan` and `asan` builds. I don't think that zeroing memory in order to make `msan` happy should prevent the other sanitizers from seeing what really happens in the code.